File: trunk/ir_web/webapps/ir_web/page-resources/js/user/forgot_password.js
/*
   Copyright 2008 University of Rochester

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

/**
 * This code is for dealing with adding and removing folders 
 * in the workspace.
 */
YAHOO.namespace("ur.login");

// action for forgot password
var savePasswordTokenAction = basePath + 'savePasswordToken.action';


YAHOO.ur.login = 
{
   /** 
     * clear out any form data messages or input
     * in the forgot password form
     */
    clearForgotPasswordForm : function()
    {
	    document.getElementById('forgotPasswordForm_email').value = "";
    },
    
    /**
     * Creates a YUI forgot password modal dialog 
     */
    createForgotPasswordDialog : function()
    {
    
	    // Define various event handlers for Dialog
	    var handleSubmit = function() {

	        YAHOO.util.Connect.setForm('forgotPasswordForm');
	    	    
	        if( YAHOO.ur.login.forgotPasswordDialog.validate() )
	        {
                var cObj = YAHOO.util.Connect.asyncRequest('post',
                savePasswordTokenAction, callback);
            }
	    };
	
		
	    // handle a cancel of the forgot password dialog
	    var handleCancel = function() 
	    {
	        YAHOO.ur.login.clearForgotPasswordForm();
	        YAHOO.ur.login.forgotPasswordDialog.hide();
	    };
	
	    var handleSuccess = function(o) 
	    {
			// check for the timeout - forward user to login page if timout
	        // occured
	        if( !urUtil.checkTimeOut(o.responseText) )
	        {       	    
	            //get the response from adding a password token
	            var response = o.responseText;
	            var forgotPasswordForm = document.getElementById('forgotPasswordDialogFields');
	    
	            // update the form fields with the response.  This updates
	            // the form, if there was an issue, update the form with
	            // the error messages.
	            forgotPasswordForm.innerHTML = o.responseText;
	    
	            // determine if the add/edit was a success
	            var success = document.getElementById("forgotPasswordForm_success").value;
	            var message = document.getElementById("forgotPasswordForm_message").value;
	  
	            //if the password token not added then show the user the error message.
	            // received from the server
	            if( success == "false" )
	            {
	                var emailError = document.getElementById('emailError');
                    emailError.innerHTML = '<p id="forgotPasswordForm_emailError">' + message + '</p>';
   
                    YAHOO.ur.login.forgotPasswordDialog.showDialog();
	            }
	            else
	            {
	                // we can clear the form if the password token was added
	                YAHOO.ur.login.clearForgotPasswordForm();
	                YAHOO.ur.login.forgotPasswordDialog.hide();
	            }
	        }
	    };
	
	    // handle form sbumission failure
	    var handleFailure = function(o)
	    {
	        alert('Email submission failed ' + o.status);
	    };

	    // Instantiate the Dialog
	    // make it modal - 
	    // it should not start out as visible - it should not be shown until 
	    // forgot password link is clicked.
	    YAHOO.ur.login.forgotPasswordDialog = new YAHOO.widget.Dialog('forgotPasswordDialog', 
        { width : "500px",
		  visible : false, 
		  buttons : [ { text:'Submit', handler:handleSubmit, isDefault:true },
					  { text:'Cancel', handler:handleCancel } ]
		} );
		
		/**
		 *  show and center the dialog
		 */
		YAHOO.ur.login.forgotPasswordDialog.showDialog = function()
		{
		    YAHOO.ur.login.forgotPasswordDialog.center();
		    YAHOO.ur.login.forgotPasswordDialog.show();
		}
   
 	    // Validate the email id
	    YAHOO.ur.login.forgotPasswordDialog.validate = function() {
	        var email = document.getElementById('forgotPasswordForm_email').value;
		    if (email == "" || email == null) 
		    {
		        alert('Email Id must be entered');
			    return false;
		    } 
		    else 
		    {
			    if (!urUtil.emailcheck(email)) 
			    {
				    alert('Invalid email address');
				    return false;
			    }
			    return true;
		    }
	    };

	    // Wire up the success and failure handlers
	    var callback = { success: handleSuccess, failure: handleFailure };
			
	    // Render the Dialog
	    YAHOO.ur.login.forgotPasswordDialog.render();
    },
    
    /**
     * Forgot the password
     */
    forgotPassword : function()
    {
	        YAHOO.ur.login.forgotPasswordDialog.showDialog();
    },
    
    init : function()
    {
        YAHOO.ur.login.createForgotPasswordDialog();
    }
}

// initialize the code once the dom is ready
YAHOO.util.Event.onDOMReady( YAHOO.ur.login.init); 

File: tags/ir_plus-2.0.8/ir_web/src/edu/ur/ir/web/action/user/ForgotPassword.java
/**  
   Copyright 2008 University of Rochester

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/  


package edu.ur.ir.web.action.user;

import org.apache.log4j.Logger;

import com.opensymphony.xwork2.ActionSupport;

import edu.ur.ir.user.IrUser;
import edu.ur.ir.user.UserService;

/**
 * Action to handle forgot password 
 * 
 * @author Sharmila Ranganathan
 *
 */
public class ForgotPassword extends ActionSupport {

	/** Eclipse generated Id */
	private static final long serialVersionUID = 7491973408867996093L;

	/**  Logger for add user action */
	private static final Logger log = Logger.getLogger(ForgotPassword.class);

	/** Email id to send the password details */
	private String email;

	/** User service */
	private UserService userService;
	
	/** Indicates whether the password token is added or not. */
	private boolean added = false;

	/** Error message to the user  */
	private String message;

	/**
	 * Execute method
	 * 
	 */
	public String execute(){
		log.debug("Execute called");
		added = false;
		
		IrUser user = userService.getUserByEmail(email); 
		
		if (user == null)
		{
			message = getText("emailDoesnotExist");
			addFieldError("emailDoesnotExist", 
					"The Email id doesnot exist in the system" + email);
		} else {
			String token = userService.savePasswordToken(user);
			userService.sendEmailForForgotPassword(token, email);
			added = true;
		}
		
		return SUCCESS;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public UserService getUserService() {
		return userService;
	}

	public void setUserService(UserService userService) {
		this.userService = userService;
	}

	public boolean isAdded() {
		return added;
	}

	public void setAdded(boolean added) {
		this.added = added;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}



}
